test/manager/upload_and_share_file_manager_test.dart

```diff
@@ -0,0 +1,166 @@
+import 'package:domain/domain.dart';
+import 'package:get_it/get_it.dart';
+import 'package:linshare_flutter_app/presentation/manager/upload_and_share_file/upload_and_share_file_manager.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:redux/redux.dart';
+import 'package:testshared/testshared.dart';
+
+import '../../domain/test/fixture/test_fixture.dart';
+import '../../domain/test/mock/repository/authentication/mock_credential_repository.dart';
+import '../../domain/test/mock/repository/authentication/mock_document_repository.dart';
+import '../../domain/test/mock/repository/authentication/mock_token_repository.dart';
+import '../fixtures/test_redux_module.dart';
+
+void main() {
+  final getIt = GetIt.asNewInstance();
+  TestReduxModule(getIt);
+
+  group('upload_and_share_file_manager_test', () {
+    MockDocumentRepository documentRepository;
+    MockTokenRepository tokenRepository;
+    MockCredentialRepository credentialRepository;
+    UploadFileInteractor uploadFileInteractor;
+    ShareDocumentInteractor shareDocumentInteractor;
+    UploadShareFileManager uploadShareFileManager;
+    UploadTaskId uploadTaskId;
+
+    setUp(() {
+      documentRepository = MockDocumentRepository();
+      tokenRepository = MockTokenRepository();
+      credentialRepository = MockCredentialRepository();
+      uploadTaskId = UploadTaskId('upload_task_id_1');
+      uploadFileInteractor = UploadFileInteractor(
+        documentRepository,
+        tokenRepository,
+        credentialRepository,
+      );
+      shareDocumentInteractor = ShareDocumentInteractor(documentRepository);
+
+      uploadShareFileManager = UploadShareFileManager(
+        getIt.get<Store<AppState>>(),
+        uploadFileInteractor,
+        shareDocumentInteractor,
+      );
+    });
+
+    void mockPermanentUploadData() {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl())
+          .thenAnswer((_) async => linShareBaseUrl);
+    }
+
+    test(
+      'justUpload just mapping result into action, then dispatch to store',
+      () async {
+        mockPermanentUploadData();
+        when(documentRepository.upload(
+          fileInfo1,
+          permanentToken,
+          linShareBaseUrl,
+        )).thenAnswer(
+          (_) async => FileUploadState(
+            Stream.fromIterable([
+              Right<Failure, Success>(fileUploadProgress10),
+              Right<Failure, Success>(fileUploadProgress100),
+              Right<Failure, Success>(FileUploadSuccess(document)),
+            ]),
+            uploadTaskId,
+          ),
+        );
+
+        await uploadShareFileManager.justUpload(fileInfo1);
+
+        expect(
+          getIt.get<Store<AppState>>().state.uploadFileState.viewState,
+          equals(Right(FileUploadSuccess(document))),
+        );
```

Comment:
it is better if we can get the all states of upload

Reply:
Store<State> is not a stream :( After `await`, the last state is hold in state '_'